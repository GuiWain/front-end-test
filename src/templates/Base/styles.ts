import styled from 'styled-components'

import UseFooter from 'components/Footer'
import UseMenu from 'components/Menu'

export const Menu = styled(UseMenu)``
export const Footer = styled(UseFooter)``

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  margin: 8rem auto 0;
  max-width: 100%;
`
export const Content = styled.div`
  max-width: 100%;
`
