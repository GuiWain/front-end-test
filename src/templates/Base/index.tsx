import * as S from './styles'

export type BaseTemplateProps = {
  children: React.ReactNode
}

const Base = ({ children }: BaseTemplateProps) => {
  return (
    <S.Wrapper>
      <S.Menu />

      <S.Content>{children}</S.Content>

      <S.Footer />
    </S.Wrapper>
  )
}

export default Base
