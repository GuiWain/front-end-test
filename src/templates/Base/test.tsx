import { render, screen } from 'utils/test-utils'

import Base from '.'

jest.mock('components/Menu', () => ({
  __esModule: true,
  default: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="Mock Menu">{children}</div>
  }
}))

jest.mock('components/Footer', () => ({
  __esModule: true,
  default: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="Mock Footer">{children}</div>
  }
}))

describe('<Base />', () => {
  it('should render the Base', () => {
    const { container } = render(
      <Base>
        <div>
          <h1>Heading</h1>
        </div>
      </Base>
    )

    expect(screen.getByText('Heading')).toBeInTheDocument()
    expect(screen.getByTestId(/Mock Menu/i)).toBeInTheDocument()
    expect(screen.getByTestId(/Mock Footer/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
