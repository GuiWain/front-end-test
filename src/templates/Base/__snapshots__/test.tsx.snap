// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Base /> should render the Base 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 8rem auto 0;
  max-width: 100%;
}

.c1 {
  max-width: 100%;
}

<div
  class="c0"
>
  <div
    data-testid="Mock Menu"
  />
  <div
    class="c1"
  >
    <div>
      <h1>
        Heading
      </h1>
    </div>
  </div>
  <div
    data-testid="Mock Footer"
  />
</div>
`;
